### PR TITLE
UI changes to accept SAML single logout requests from the federated IdP

### PR DESCRIPTION
**Pull request by Isanka95**
### Proposed changes in this pull request

- Resolves: wso2/product-is#5717

- Removed the **Enable Logout** configuration and put it under the new **Single Logout Profile** configuration.

-  Added another new checkbox to accept single logout requests from federated IdP under the single logout profile configuration.

- Changed the description of the Enable Logout checkbox.

**Previous appearance of the identity provider registration page**

Screenshot from 2019-11-15 22-18-24

**New appearance**

Screenshot from 2019-11-15 22-54-57

### CODE REVIEW

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 1]
```diff
@@ -119,6 +119,7 @@
     boolean isSLOEnabled = false;
     boolean isLogoutRequestSigned = false;
     String logoutUrl = null;
+    boolean isSLORequestAccepted =false;
```
Comment (sachiniWettasinghe) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: formating issue

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 2]
```diff
@@ -995,6 +1001,14 @@
     if (StringUtils.isBlank(logoutUrl)) {
         logoutUrl = StringUtils.EMPTY;
     }
+
+    String sloRequestAcceptedChecked = "";
+    if (identityProvider != null) {
+        if (isSLORequestAccepted) {
```
Comment (sachiniWettasinghe) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: improve this to `if (identityProvider != null && isSLORequestAccepted)`

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 3]
```diff
@@ -1589,6 +1589,15 @@ private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp
         property.setValue(paramMap.get("logoutUrl"));
         properties.add(property);
 
+        property = new Property();
+        property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_SLO_REQUEST_ACCEPTED);
+        if ("on".equals(paramMap.get("sloRequestAccepted"))) {
+            property.setValue("true");
+        } else {
+            property.setValue("false");
+        }
```
Comment (janakamarasena) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java:1598]: Lets move these to constants

Comment (IsankaSR) [reply components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java:1598]: Fixed with PR https://github.com/wso2/carbon-identity-framework/pull/3738

**Pull request merged**